etcdserver/api/v3rpc/auth.go

```diff
@@ -68,8 +68,7 @@ func (as *AuthServer) RoleGrant(ctx context.Context, r *pb.RoleGrantRequest) (*p
 }
 
 func (as *AuthServer) UserAdd(ctx context.Context, r *pb.UserAddRequest) (*pb.UserAddResponse, error) {
-	plog.Info("not implemented yet")
-	return nil, nil
+	return as.authenticator.UserAdd(ctx, r)
```

Comment:
At rpc level, we have to convert all errors into the error defined by the rpc layer. Or the client side will not be able to parse the error.